boards/DCM/Test/Src/Test_StateMachine.cpp

```diff
@@ -580,4 +670,171 @@ TEST_F(
         0.0f, App_CanTx_GetPeriodicSignal_TORQUE_REQUEST(can_tx_interface));
 }
 
+// DCM21
+TEST_F(DcmStateMachineTest, inverter_switches_closed_when_AIRS_closed)
+{
+    // Both inverter switches are open to begin with
+    ASSERT_EQ(turn_on_right_inverter_fake.call_count, 0);
+    ASSERT_EQ(turn_on_left_inverter_fake.call_count, 0);
+
+    // Close AIR-, both inverter switches should still be open
+    App_CanRx_BMS_AIR_STATES_SetSignal_AIR_NEGATIVE(
+        can_rx_interface, CANMSGS_BMS_AIR_STATES_AIR_NEGATIVE_CLOSED_CHOICE);
+    LetTimePass(state_machine, 10);
+    ASSERT_EQ(turn_on_right_inverter_fake.call_count, 0);
+    ASSERT_EQ(turn_on_left_inverter_fake.call_count, 0);
+
+    // Close AIR+, both inverter switches should now be closed
+    App_CanRx_BMS_AIR_STATES_SetSignal_AIR_POSITIVE(
+        can_rx_interface, CANMSGS_BMS_AIR_STATES_AIR_POSITIVE_CLOSED_CHOICE);
+    LetTimePass(state_machine, 10);
+    ASSERT_EQ(turn_on_right_inverter_fake.call_count, 1);
+    ASSERT_EQ(turn_on_left_inverter_fake.call_count, 1);
+
+    // Left and right inverter should be closed once
+    is_right_inverter_on_fake.return_val = true;
+    is_left_inverter_on_fake.return_val  = true;
+    LetTimePass(state_machine, 10);
+    ASSERT_EQ(turn_on_right_inverter_fake.call_count, 1);
+    ASSERT_EQ(turn_on_left_inverter_fake.call_count, 1);
+}
+
+TEST_F(DcmStateMachineTest, init_to_fault_state_on_left_inverter_fault)
+{
+    EXPECT_EQ(
```

Comment:
We should always be in the init state here right? For cases like this do you think it would be better to do:

- Introduce fault signal
- let time pass for 9ms 
- Check if were still in init
- Let time pass for 1ms
- Check for fault

Reply:
done